### DISCUSSION TITLE
Adding a Permissive license to Slint + other changes

### DISCUSSION THREAD

**Discussion by aurindam** [slint-ui/slint#2706] [General] [answered]
**Goal**: We welcome your feedback on the proposed licensing changes as described below. After 24th May, we will summarise the discussion.

- [Adding a royalty-free (permissive) license as a third option](#adding-a-royalty-free-(permissive)-license-as-a-third-option)
- [Simplifying the CLA](#simplifying-the-cla)
- [Changing the license of examples and demos from GPLv3 to MIT](#changing-the-license-of-examples-and-demos-from-gplv3-to-mit)

## Adding a royalty-free (permissive) license as a third option

Currently, Slint is available under GPLv3 and the Slint proprietary License (which has a free "ambassador" tier). Since the 1.0 release, we've received over 100 Ambassador applications!

Spurred by the positive response we received from the community, we'd like to **add** a more permissive license to the current license options. The current Ambassador license granting process involves [digitally signing the proprietary license agreement][#1]. Some of the terms and conditions in the proprietary license agreement are relevant for paying customers but not for Ambassadors. In the future, we want to make this process simpler by introducing a more permissive license. Unfortunately none of the [OSI Approved Licenses](https://opensource.org/licenses/) fit our needs and hence a new license. The salient features of this new license would be:

- More permissive than GPLv3
- Attribution to use of Slint
- For building any desktop or web (wasm) application

After this change, Slint will be available under three licenses, at your option

1. GPLv3 (unchanged)
2. (the new) Royalty-free License
3. Proprietary license (with the ambassador terms no longer needed)

The Royalty-free License will be added to the Slint repository so that users can use this License without having to fill up a form to apply for it.

Current Ambassador licensees can continue using Slint under the Ambassador license issued to them since the license is perpetual.

Below is the draft text of the Royalty-free License:

> # Slint Royalty-free Desktop and Web Applications License
>
> ## Preamble
>
> Slint is a toolkit that can be used to build user interfaces for applications. Slint (hereafter referred to as "Software") is made available under different licenses by SixtyFPS GmbH incorporated at Oranienburger Str. 44, 16540 Hohen Neuendorf, Germany ("SixtyFPS"). The "Slint Royalty-free Desktop and Web Applications License" is suitable for those who develop desktop or web applications and do not want to use open source components under copyleft licenses.
>
> ### 1. Grant of Rights
>
> You are hereby granted a world-wide, royalty-free, non-exclusive license to use, reproduce, make available, modify, display, perform, distribute the Software as part of a Desktop or Web Application.
>
> A Desktop Application is a computer program that is designed to run on a general-purpose computer (PC or notebook), typically installed and executed locally on the computer's operating system. A Web Application is a computer program that is running on a server and accessed using a web browser or client program.
>
> Desktop Application and Web Application are hereafter referred to as "Application".
>
> ### 2. Limitations
>
> The license does not permit to distribute or make the Software publicly available alone and without integration into an Application. For this purpose you may use the Software under the GNU General Public License, version 3.
>
> The license does not permit the use of the Software within Embedded Systems. An Embedded System is a computer system designed to perform a specific task within a larger mechanical or electrical system. For the purposes of this license, mobile phones are not considered to be Embedded Systems.
>
> ### 3. License Conditions - Attribution
>
> You may distribute the Software as part of an Application, modified or unmodified, provided that You do all of the following:
>
> 1. Display the [`AboutSlint`][#2] widget in an "About" screen or dialog that is accessible from the top level menu of the Application.
>
> 2. Display the [Slint attribution badge][#3] on a public webpage, where the binaries of your Application can be downloaded from, in such a way that it can be easily found by any visitor to that page.
>
> 3. You may not remove or alter any license notices (including copyright notices, disclaimers of warranty, or limitations of liability) contained within the source code form of the Software.
>
>  By distributing the Software as part of an Application, you give us consent to use your name, logo and your Application to be showcased as "Made with Slint".
>
> ### 4.  Warranty and Liability
>
> SixtyFPS is only liable for conflicting rights of third parties if SixtyFPS was aware of these rights without informing you. Unless required by applicable law or agreed to in writing, SixtyFPS provides the SOFTWARE on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied, including, without limitation, any warranties or conditions of MERCHANTABILITY, or FITNESS FOR A PARTICULAR PURPOSE.
>
> UNLESS REQUIRED BY LAW, SIXTYFPS WON'T BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL DAMAGES ARISING IN ANY WAY OUT OF THE USE OF THE SOFTWARE.

## Simplifying the CLA

We intend to replace our existing CLA with a simpler notice that would ensure that contributions are licensed under [MIT No Attribution License][#4]. This gives us the flexibility to adapt the licensing terms for Slint while giving you the flexibility to use your code in other projects under licenses of your choice.

Contributors who have already signed the current CLA ~~do not~~ need to sign the new CLA since the CLA-assistant bot [automatically asks users to re-sign the CLA for each new pull request in the event the associated Gist & CLA has changed](https://github.com/cla-assistant/cla-assistant#contributor-license-agreement-cla-assistant) 

Below is the draft text of the new CLA:

> # Slint Individual Contributor License Agreement Version 2.0
>
> You hereby license all present, past, and future contributions to Slint, represented by SixtyFPS GmbH incorporated at Oranienburger Str. 44, 16540 Hohen Neuendorf, Germany ("We" or "Us"), under the terms of the [MIT No Attribution License][#4]:
>
> ----
>
> MIT No Attribution
>
> Copyright **YEAR** **COPYRIGHT HOLDER**
>
> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so.
>
> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
>
> ----
>
> This gives us the flexibility to adapt the licensing terms for Slint while giving you the flexibility to use your code in other projects under licenses of your choice.
>
> Please note that you must have written the contribution 100% yourself and that no rights have been transferred to third parties (e.g. your employer). In any other case, please let us know.
>
> We believe that open source software development and communities are the foundation for a healthy ecosystem of high-quality software, where everyone can learn, improve and give back. We commit to upholding this foundation and pledge by promising to continue to develop Slint in the open under an open source license, as approved by the [Open Source Initiative][#5].

For comparison, you can see our current CLA [here][#6]

## Changing the license of examples and demos from GPLv3 to MIT

As we add more examples and demos, it is natural to expect that users would make use of them to get started with Slint. Currently these are licensed under GPLv3 which could be a problem for users who want to license their project under non-GPL compatible license. Hence, we're considering changing the license to a more permissive license, specifically the [MIT license][#7].

> MIT License
>
> Copyright **YEAR** **COPYRIGHT HOLDER**
> 
> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
> 
> The above copyright notice and this permission notice (including the next paragraph) shall be included in all copies or substantial portions of the Software.
> 
> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 

 

[#1]: https://slint-ui.com/ambassador-program.html#application
[#2]: https://slint-ui.com/snapshots/master/docs/slint/src/builtins/widgets.html#aboutslint
[#3]: https://user-images.githubusercontent.com/959326/237060414-71f0e7e5-f1d9-42ee-9a88-d28282cf5f52.png
[#4]: https://spdx.org/licenses/MIT-0.html
[#5]: https://opensource.org
[#6]: https://cla-assistant.io/slint-ui/slint
[#7]: https://spdx.org/licenses/MIT.html

**Discussions on other threads**

1. https://www.reddit.com/r/rust/comments/13fexq3/feedback_requested_slint_declarative_gui_toolkit/
2. https://www.reddit.com/r/cpp/comments/13fezyh/feedback_requested_slint_declarative_gui_toolkit/
3. https://www.reddit.com/r/programming/comments/13ff0ek/feedback_requested_slint_declarative_gui_toolkit/
4. https://www.reddit.com/r/opensource/comments/13feyfe/feedback_requested_slint_declarative_gui_toolkit/

**Comment by Be-ing** [comment]
Overall, I think this is a decent idea. However, I suggest to not use the term "permissive license" because that phrase typically refers to lax non-copyleft licenses like MIT and Apache licenses. Perhaps "royalty-free promotional license" would be better? Or some phrasing along those lines

Also, the definition of an "Embedded System" in the proposed license seems overly vague, which some people might interpret as prohibiting hobbyist usage on a Raspberry Pi. I suggest defining that to focus on *selling* hardware products containing Slint. (Tangentially, I'm not a big fan of the GPLv3's definition of "User Product" which I don't think is quite what you're going for here, as "Embedded System" could cover physical products that are primarily/exclusively sold to businesses rather than consumers.)

**Reply by aurindam** [reply]
One alternative would be to offer a free tier (under the proprietary license) for hobby/academic projects using Slint on "Embedded System".

**Reply by ahayzen-kdab** [reply]
This would then mean there would be 4 (?) different possible licenses to choose from, having 3 already seemed potentially confusing and might need something like the CC Choose page https://creativecommons.org/choose/

**Reply by aurindam** [reply]
There would be 3 license options - GPLv3, Royalty-free (desktop and web) and proprietary.

The proprietary license will have different pricing options - free, per-user and per-project (similar to what is offered currently)

**Reply by Be-ing** [reply]
What's the point of having both the royalty-free license and a no cost proprietary license?

**Comment by jturcotte** [comment]
A few questions about the Royalty-Free license:

- **"You are hereby granted"**
    - What happens if the application is open-source (e.g. under MIT), forked by a different person and then redistributed?
- **"distribute the Software as part of a Desktop or Web Application"**
    - Does this mean as binary, or also as source? For the source distribution it would be interesting to know what happens for modifications to it, and to what license(s) can be used for redistribution of copies using those modifications.
- **"provided that You do all of the following"**
    - For how long? What happens to previous and future versions of the software if any of those conditions becomes unmet?
- **"If your Application has a public page"**
    - If many are available, on any of them? For example if a project has both a GitHub and web page.
- **"Give us consent to use your name, logo, and your Applications in our marketing materials by filling up a form on Slint website."**
    - This leaves some room to imagine all of what this could mean. If you can define more precisely what "marketing materials" means, without sounding too much like there is an open end to usages on your side provided in the exchange for the license, you will possibly get a better adoption.
    - Maybe you could define somewhere what would be the conditions and process to revert the publication of this information, if possible, as this isn't in the licensee's control.
    - Does this needs to be done for every application a person/company uses Slint for? If so, what defines if it's a different application or not?
    
I surprisingly find the legal text quite clear, but consider still putting a TL;DR to explain the conditions as concisely as possible in a few bullet points when introducing this option to reduce misunderstandings that could result in FUD.

**Reply by aurindam** [reply]
Thanks for the feedback and suggestions. Indeed, an FAQ section / TL;DR should be added.

> What happens if the application is open-source (e.g. under MIT), forked by a different person and then redistributed?

The Royalty-free license does not restrict users on how they license their Application. In the above example, the user may choose to use MIT-license for their Application, which can be forked by a different person and then redistributed.

> Does this mean as binary, or also as source? For the source distribution it would be interesting to know what happens for modifications to it, and to what license(s) can be used for redistribution of copies using those modifications.

Both binary and source distributions are permitted. However, §2 Limitations states "The license does not permit to distribute or make the Software publicly available alone and without integration into an Application. For this purpose you may use the Software under the GNU General Public License, version 3." -> so source distribution (including modifications) is possible but only under GPLv3. 

> For how long? What happens to previous and future versions of the software if any of those conditions becomes unmet?

As long as the Application is built with Slint under the Royalty-free license.

> If many are available, on any of them? For example if a project has both a GitHub and web page.

Perhaps only on the main/primary page of the Application? If a project has both a GitHub and web page, perhaps on the GitHub page?

> This leaves some room to imagine all of what this could mean. If you can define more precisely what "marketing materials" means, without sounding too much like there is an open end to usages on your side provided in the exchange for the license, you will possibly get a better adoption.

The intention is to create a Slint showcase like https://madewithvuejs.com , https://madewithtailwindcss.com etc.

> Maybe you could define somewhere what would be the conditions and process to revert the publication of this information, if possible, as this isn't in the licensee's control.

Fair point. We will add some info around that in the form submission page.

> Does this needs to be done for every application a person/company uses Slint for? If so, what defines if it's a different application or not?

As the intention is to create a Slint showcase page, submission per Application would be preferable. But would this be a blocker?

**Reply by jturcotte** [reply]
> The Royalty-free license does not restrict users on how they license their Application. In the above example, the user may choose to use MIT-license for their Application, which can be forked by a different person and then redistributed.

I was wondering what happens with the Royalty-Free license for the forked application. My understanding is that the license only applies to name+application, and that if anybody else forks the code and wants to distribute binaries under MIT without providing their modifications they have to re-apply for a Royalty-Free license even if it's the "same" application.

> As long as the Application is built with Slint under the Royalty-free license.

I mean that the about box stays with the application binaries, so if I put my binaries on the app store, the binaries keep a valid license as long as a new version of the binary doesn't remove Slint from the about box.

But the project's home page is a different thing, it can be updated separately. So if I accidentally or voluntarily remove the "built with Slint" badge" and breach the contract, does it invalidates the Royalty-Free license only for future binaries, or both for future and previously built binaries? If I stretch the logic further, could I add the badge, build my application, push it to the app store, and remove the badge from the main page the day after and still fulfill my part of the contract?

Sorry for nitpicking, this probably wouldn't matter too much in practice, but just wanted to mention this ambiguity.
I think one way that this could be agreed would be to tie the badge to each version and that the badge (or text) needs to be added to, or near, the page where the binaries of the licensed version are downloaded from. So this would be on the App Store **and** the GitHub release pages if the binaries are released from both for desktop and iOS.

> As the intention is to create a Slint showcase page, submission per Application would be preferable. But would this be a blocker?

I don't personally have an opinion, I just think that it would be important to define when the form needs to be filled again as there is no approval process for the license. Users will try their best to fulfill the license and just hope that they have the right to distribute binaries of their application afterwards without risking being sued. One example is if I release an iOS app using Slint as "Temple Fun" and then again but with slightly different app named "Temple Fun 2" or "Temple Fun Light", would this require filling the form again, or not?

I understand that it's part of the deal and this is where your company is getting value out of it, but I feel like the form filling part makes the license validity more difficult to ensure. I don't know if there is a way that you could make the consent a "pull" operation instead of "push"? For example that the application puts the text "This application is using the Slint Royalty-Free License" somewhere (near the badge or in the about box), and this would give you the right, per the license conditions, to use the application and company name in your marketing material without asking first. This way the applications could fork, rename or change owner and they wouldn't need to worry at all about all the license check boxes being green or that one of them is suddenly red, as long as they keep this mention.

You could still ask them to register or tag you on Twitter if they release the app so that you know about it, but at least their license wouldn't depend on you knowing about the app, it would be tied to their consent for you to use their name even if you don't know it yet.

Thank you for the rest of the clarifications.

**Reply by aurindam** [reply]
> I was wondering what happens with the Royalty-Free license for the forked application. My understanding is that the license only applies to name+application, and that if anybody else forks the code and wants to distribute binaries under MIT without providing their modifications they have to re-apply for a Royalty-Free license even if it's the "same" application.

You are right. What I should have added is that this license will be added to the repo so that it's "auto-granted" and there is no need to "apply" for the license in the same way that one does not need to apply for the GPL. I will update the text to clarify this point.

>  think one way that this could be agreed would be to tie the badge to each version and that the badge (or text) needs to be added to, or near, the page where the binaries of the licensed version are downloaded from.

Something like - "Display the [Slint attribution badge](https://user-images.githubusercontent.com/959326/237060414-71f0e7e5-f1d9-42ee-9a88-d28282cf5f52.png) on a public page where the binaries of your Application can be downloaded in such a way that it can be easily found by any visitor to that page." ?

>  I understand that it's part of the deal and this is where your company is getting value out of it, but I feel like the form filling part makes the license validity more difficult to ensure. I don't know if there is a way that you could make the consent a "pull" operation instead of "push"? 

There are tools that can check backlinks and perhaps something like that could be used to identify which Applications are built with Slint. Regarding the consent part, I would lean towards being conservative and asking for it explicitly.
Proposed text - "Give us consent to use your name, logo, and your Applications to be showcased as a "Made with Slint""

**Reply by jturcotte** [reply]
> Something like - "Display the [Slint attribution badge](https://user-images.githubusercontent.com/959326/237060414-71f0e7e5-f1d9-42ee-9a88-d28282cf5f52.png) on a public page where the binaries of your Application can be downloaded in such a way that it can be easily found by any visitor to that page." ?

Sounds good to me.

**Reply by jturcotte** [reply]
> this license will be added to the repo so that it's "auto-granted" and there is no need to "apply" for the license in the same way that one does not need to apply for the GPL.

I do see it as "applying" if having a valid license requires filling the form to grant explicit consent for any combination name+application used to distribute the application binaries using Slint. But OK at least I guess there is no approval process involved, it's kind of push-and-forget.

**Reply by aurindam** [reply]
> I do see it as "applying" if having a valid license requires filling the form to grant explicit consent for any combination name+application used to distribute the application binaries using Slint.

Removed the 'form filling' part and replaced it with "Give us consent to use your name, logo, and your Applications to be showcased as a "Made with Slint""

**Reply by jturcotte** [reply]
Ah I just saw that you proposed to remove the form from the text in your previous message. This looks good I think, thank you.

Maybe it could be specified how the consent can be given outside the license itself.

**Comment by hunger** [comment]
I really like the direction, but I think section 3.3 could use a bit more elaboration:

* When do I need to register? Before I push Slint code? Before I ship (which will be very close to "once I use SLint" for projects working in the open)? Anytime in between?

* Do I need to unregister when I no longer use Slint? Especially if I am asked to register early (e.g. before pushing a patch adding Slint), we could have a lot of false positives.

**Reply by aurindam** [reply]
> When do I need to register? Before I push Slint code? Before I ship (which will be very close to "once I use SLint" for projects working in the open)? Anytime in between?

Since the goal is to create a Showcase page, the submission (or registration) should be when the Application is "ready" or "released".

> Do I need to unregister when I no longer use Slint? Especially if I am asked to register early (e.g. before pushing a patch adding Slint), we could have a lot of false positives.

@jturcotte also brought up this point. We should provide info on how to remove a submission from the Showcase page

**Reply by aurindam** [reply]
Update: The form submission part is now removed from the license text.

**Comment by JohnAZoidberg** [comment]
Is there are reason for using GPL instead of a file-based copy-left license like MPL2.0?
On embedded systems using Slint would cause the entire firmware to be classified as a derivative work, making it GPL.

Why doesn't the new license apply to embedded systems?

Would a UEFI app be considered application or embedded? I could argue both ways.

**Reply by aurindam** [reply]
> Is there are reason for using GPL instead of a file-based copy-left license like MPL2.0?

We had considered the MPL2.0 but dropped it due to its weak copy-left nature.

> Why doesn't the new license apply to embedded systems?

As most of the Slint Ambassadors are building desktop applications, we wanted to introduce this license to cover desktop and web applications first. 

> Would a UEFI app be considered application or embedded? I could argue both ways.

This is a tricky one (one of the few corner cases). 
If the UEFI app is run on a general-purpose computer, then it can be considered as an application. 
If the UEFI app is run on a computer system designed to perform a specific task within a larger mechanical or electrical system, then it's embedded. 
In other words, "how" and "where" the UEFI app is used determines whether it should be considered as application or embedded.

**Reply by patstew** [reply]
Is the intention to allow android/ios when slint supports them? They don't seem to fall into the 'Desktop' or 'Embedded' categories.

The above licence seems to me to be weaker copy-left than MPL. Since your choices are GPL or new licence, and you explicitely can't release Slint alone under the new licence, it looks like you're pushing anyone who doesn't want to/can't go full GPL towards a fully closed release, where they don't release any source at all. At least with MPL they'd have to release any modifications to Slint. You could also modify MPL, e.g. to enforce that any released modifications to Slint have to be MIT licenced rather than MPL, and you could add your attribution clause.

**Reply by aurindam** [reply]
> Is the intention to allow android/ios when slint supports them?

Yes, if there is enough interest when Slint supports mobile platforms.

> it looks like you're pushing anyone who doesn't want to/can't go full GPL towards a fully closed release, where they don't release any source at all.

While this license is for aimed at those who doesn't want to/can't go full GPL, the license neither places any restrictions on the license of the Application nor on the license of the modifications to Slint. So, its up to the user to decide which license they will use.
Say for example, Alice uses Slint under this new license to develop Application A and modifies Slint in some way. Alice can decide to release Application A under MIT license. She can also decide to release the modifications to Slint under MIT license. The other end of the spectrum would be that Alice decides to neither release any source of Application A nor the modifications to Slint. 

Note: The license also permits distribution of source code of Slint but only under GPLv3.

**Reply by patstew** [reply]
Fair enough, it just seems odd to reject MPL because it's weak copyleft but then allow no copyleft at all. If I were you I'd have an MPL/LGPL style licence, or a similar one that allows you to incorporate the changes into your commercial version, so that you're either getting code or money from people using it.
In any case I'm pleased you're planning on this kind of licence, it makes slint a consideration for our company. We currently use Qt, and occasionaly it annoys me enough to consider using something else next time. but realistically we're very small and we're not going to tie ourselves to something we have to pay for, and it wouldn't be worth investing in a new stack for a one-off intro licence. We're quite happy to contribute to things we use though. A big company might be a lot less willing to contribute to open source, but much more willing to splash some cash to avoid that if there was no option.

**Reply by vorot93** [reply]
@aurindam Would you consider adding a Classpath exception to GPL if MPL is too weak for Slint? It's essentially like LGPL, but for static-compiled applications, allowing for distribution of non-GPL software linked to GPL.

**Comment by nicoburns** [comment]
I like the general direction here, but I think the split between "application" and "embedded" needs to be clarified in both directions:

- Unless the intention is not to include phone (e.g. android/ios) apps, I would like to see "mobile phones" and "tablets" explicitly included in the grant along with "desktop" and "web".
- On the other hand, I understand that tablets in particular can be used as part of embedded devices, and this presumably shouldn't be covered. And I would note that a web app could potentially be used for this purpose too. I think you'll need to define this quite carefully. For example, how would one classify a game console? Would that be different if Slint was used for a single "app/game" on the console vs. for the built-in system UI?

I am also wondering how modifications to Slint itself are covered under this licence. As-is, it seems that users of this new licence would not only not be required to, but would actually not be allowed to distribute the source of any modifications to Slint itself. Seems to me that something MPL-like, where modifications to Slint must be contributed back but not applications built on top might be good.

**Reply by aurindam** [reply]
> Unless the intention is not to include phone (e.g. android/ios) apps.

Its currently not included as Mobile support is "TODO" https://github.com/slint-ui/slint#current-status 

> As-is, it seems that users of this new licence would not only not be required to, but would actually not be allowed to distribute the source of any modifications to Slint itself. 

The license does not restrict 'if' and 'how' the modifications to Slint should be distributed. Say for example, Alice uses Slint under this new license to develop Application A and modifies Slint in some way. Alice can decide to release Application A under MIT license. She can also decide to release the modifications to Slint under MIT license. The other end of the spectrum would be that Alice decides to neither release any source of Application A nor the modifications to Slint.

Note: The license also permits distribution of source code of Slint but only under GPLv3.

> Would that be different if Slint was used for a single "app/game" on the console vs. for the built-in system UI?

Yes, you are right.

**Comment by mahmoudajawad** [comment]
I'm glad you kept GPL license as it indicates helps contributors know their contributions will never be blocked behind a proprietary project because someone decided to fork the project and not share changes back.

**Reply by Be-ing** [reply]
That's covered by this clause of the CLA:

> We believe that open source software development and communities are the foundation for a healthy ecosystem of high-quality software, where everyone can learn, improve and give back. We commit to upholding this foundation and pledge by promising to continue to develop Slint in the open under an open source license, as approved by the [Open Source Initiative](https://opensource.org/).

which was added after discussion in #244

**Comment by aurindam** [comment, marked as answer]
Based on feedback from the community and subsequent review with legal, we made some changes

### Changes to Slint Royalty-free Desktop and Web Applications License

- Rewording of 

> By distributing the Software as part of an Application, you give us consent to use your name, logo and your Application to be showcased as "Made with Slint".

to

> (d) You allow SixtyFPS to use your Application on the website and in advertising materials of SixtyFPS as a reference and to display your logo and trademark for this purpose.

to define clearly the purpose and scope of how we may refer to the Application and or display your logo and trademark.

### Changes to CLA

- Committing to provide a Royalty-free license by adding the text

> Further, we commit to provide a royalty-free license for those who develop desktop or web applications and do not want to use open-source components under copyleft licenses.

based on concerns from the community that the Royalty-free license could be revoked.
https://www.reddit.com/r/rust/comments/13fexq3/comment/jjx1o62/
https://www.reddit.com/r/rust/comments/13fexq3/comment/jjvhbml/

- Rewording of

> We commit to upholding this foundation and pledge by promising to continue to develop Slint in the open under an open source license, as approved by the [Open Source Initiative](https://opensource.org/).

to

> We commit to upholding this foundation and pledge by promising to continue to develop Slint in the open under an open-source license compliant with the [Open Source Definition](https://opensource.org/osd/).

based on legal advice that referring to the Open Source Definition is a better since [open source licenses are licenses that comply with the Open Source Definition](https://opensource.org/licenses/).

The PRs are: #2884, #2885, #2886, #2887, #2888, #2891

**Comment by bushrat011899** [comment]
> For the purposes of this license, mobile phones are not considered to be Embedded Systems.
Not a lawyer so take my opinion with a grain of salt. I feel like this should instead prohibit the use with commercial embedded devices. This would allow a hobbyist to make something for their Android Tablet (which is currently grey-area with this wording), or their Raspberry Pi (an embedded device commonly used in personal hobby projects). In general, I think you want to encourage "hobby" work as much as possible, because most enthusiasts push their hobbies into their profession.

**Reply by aurindam** [reply]
Yes indeed. We will try to come up with a solution for Hobby projects, academic projects etc. (non-commercial use)

**Comment by schnapper79** [comment]
So the possibility for companies using slint for embedded devices within the ambassador licence is being removed, leaving buyout+volume license or Gplv3? At least that's what I understand. Embedded as in using Linux for a machine.

**Reply by ogoffart** [reply]
Yes, that's correct.

Note that the existing ambassador licenses are perpetual. If you have signed the ambassador license before this change take effect, you can continue to use that license forever.

The change hasn't taken effect yet, so there is still time to get a license with the old terms if you'd like.

**Reply by aurindam** [reply]
> So the possibility for companies using slint for embedded devices within the ambassador licence is being removed, leaving buyout+volume license or Gplv3? At least that's what I understand. Embedded as in using Linux for a machine.

The Flex license can also be used to develop for embedded devices (this is still in planning stage)

**Comment by Quackdoc** [comment]
Please reword the licence for clarification.  Particularly the limitations section I find far to vague.

```
2. Limitations
The license does not permit to distribute or make the Software publicly available alone and without integration into an Application. For this purpose you may use the Software under the GNU General Public License, version 3.

The license does not permit the use of the Software within Embedded Systems. An Embedded System is a computer system designed to perform a specific task within a larger mechanical or electrical system. For the purposes of this license, mobile phones are not considered to be Embedded Systems.
```

it's hard to grasp what this actually means. both sections. 

for the first subsection, "distribute or make the Software publicly available alone and without integration into an Application." this makes very little sense. at what point is slint considered to be "part of an application" and what point is it considered to be "the application". for instance if you use marine greetdm a slint based greetd. is slint "the application" or "part of the application". the primary GUI of the application is slint however it still runs functions to interface with the OS in rust as well as having a TUI interface. would this be permissible or not? this needs more clarification on to what extent is the slint considered part of a program or not.

the second part is confusing too. "designed to perform a specific task within a larger mechanical or electrical system" mobile phones would not be considered embedded under this definition anyways since they are a singular electrical system with a broad variety of tasks however it is debatable if running an application inside of a phone would be considered an "embedded task that is provided as an exception to the licence" since it would be a singular task inside of a larger electrical system. and considering that phones are explicitly mentioned to not be considered part of an embedded system. would hand held consoles running linux and a custom UI based on slint be considered embedded systems? 

the licence does not do a good enough job at declaring the terms used for limitations. this leaves too much wiggle room for trying to bypass the limitations of the licence as well as too much room for potentially unwittingly violating the licence. and while im sure there may be context provided in this discussion for some of these, this is not a valid resource when dealing with licensing. 

regardless love the UI kit, 

Mitchel Stewart.

**Reply by aurindam** [reply]
Thanks Mitchel for the feedback. 

Replying to your specific questions:

> or instance if you use marine greetdm a slint based greetd. is slint "the application" or "part of the application". the primary GUI of the application is slint however it still runs functions to interface with the OS in rust as well as having a TUI interface. would this be permissible or not? this needs more clarification on to what extent is the slint considered part of a program or not.

As long as Slint is used in a Desktop or Web Application, the license permits the usage. In your example, marine greetdm is an Application and Slint is integrated into the Application. Hence, this is permitted.

NOTE: "The license does not permit to distribute or make the Software publicly available alone and without integration into an Application." is to prevent a user from misusing the Royalty-free license to redistribute Slint as a standalone package. 

> would hand held consoles running linux and a custom UI based on slint be considered embedded systems?

According the license, "An Embedded System is a computer system designed to perform a specific task within a larger mechanical or electrical system.".  If the hand held console running Linux does not fall under this definition, then its not considered an an embedded system.

> the licence does not do a good enough job at declaring the terms used for limitations. this leaves too much wiggle room for trying to bypass the limitations of the licence as well as too much room for potentially unwittingly violating the licence.

We will try to make this clearer in the license text.

**Reply by Quackdoc** [reply]
Thanks for the clarification. I do think the embedded part ought to be clarified more. Consider the following device;

a smart monitor which can optionally connect to a smart home ecosystem with the capabilities that can monitor, log and report temps in a single room or a series of rooms to an SD card or server. such device would fall squarely in between definitions. 

<details>
  <summary>Use Cases</summary>
  
under the following uses for this product, it would not be considered an embedded device
- a standalone device which logs to memory card and reports current temp;
single system, not part of a larger mechanical or electrical system

under which this could be considered to be an embedded device or not depending on interpretation
- a device which can connect to a remote server to serve a smart wireless ecosystem and bridge multiple smaller sensors;
multi-system but not connected via electrical or mechanical means, is it part of an electrical system in this case? leaves too much unclear and it's may be reasonable to assume this is not the case. these tasks could be considered a specific task or multiple tasks, in the latter case it would not fall under embedded.

under which it would be considered a embedded use case
- a smart thermostat which can control and report temperatures to a home furnace;
controlling temp would be a specific use connected to a larger mechanical system
- temp logging for a smart home ecosystem
___________________
</details>

I believe it would be best to 
- specify if you mean Primary or Possible use within a system (Ex: say "any system in which primary use case is to be used within..." Any system in which it's possible to connect to and serve a larger... )
- better define mechanical/electrical system and how they could be connected (Ex: "any device which is electically connected to...", "any device which can join a network of and serve the purpose of...")
- better clarify what a "Specific task" is. (Ex: is monitoring temps connecting to other devices to read temps and controlling temps a specific task? what about interactions with the device?).

clarifying these would make it clear where the device above stands in regards to whether or not it would fall under embedded usecase 

Thanks for getting back, I personally would be working strictly with gpl compatible works, but I have friends who are interested but find the wording of the licence too vague so clarification is much appreciated.
